test/LibraryManager.Test/LibraryInstallationStateTest.cs

```diff
@@ -26,5 +52,155 @@ public void FromInterface()
+        {
+            var state = new Mocks.LibraryInstallationState
+            {
+                ProviderId = "unpkg",
+                DestinationPath = "_path_",
+                Files = new List<string>() { "a", "b" },
+            };
+
+            ILibraryOperationResult result = await state.IsValidAsync(_dependencies);
+
+            Assert.IsFalse(result.Success);
+            Assert.AreEqual(result.Errors.Count, 1);
+            Assert.AreEqual(result.Errors.First().Code, "LIB006");
+        }
+
+        [TestMethod]
+        public async Task IsValidAsync_State_HasUnknownLibrary()
+        {
+            var state = new Mocks.LibraryInstallationState
+            {
+                ProviderId = "unpkg",
+                LibraryId = "_lib_",
+                DestinationPath = "_path_",
+                Files = new List<string>() { "a", "b" },
+            };
+
+            ILibraryOperationResult result = await state.IsValidAsync(_dependencies);
+
+            Assert.IsFalse(result.Success);
+            Assert.AreEqual(result.Errors.Count, 1);
+            Assert.AreEqual(result.Errors.First().Code, "LIB002");
+        }
+
+        [TestMethod]
+        public async Task IsValidAsync_State_FileSystem_LibraryIdHasInvalidPathCharacters()
+        {
+            var state = new Mocks.LibraryInstallationState
+            {
+                ProviderId = "filesystem",
+                LibraryId = "|lib_",
+                DestinationPath = "_path_",
+                Files = new List<string>() { "a", "b" },
+            };
+
+            ILibraryOperationResult result = await state.IsValidAsync(_dependencies);
+
+            Assert.IsFalse(result.Success);
+            Assert.AreEqual(result.Errors.Count, 1);
+            Assert.AreEqual(result.Errors.First().Code, "LIB002");
+        }
+
+        [TestMethod]
+        public async Task IsValidAsync_State_FileSystem_LibraryIdDoesNotExist()
+        {
+            var state = new Mocks.LibraryInstallationState
+            {
+                ProviderId = "filesystem",
+                LibraryId = "lib",
+                DestinationPath = "lib",
+            };
+
+            ILibraryOperationResult result = await state.IsValidAsync(_dependencies);
+
+            Assert.IsFalse(result.Success);
+            Assert.AreEqual(result.Errors.Count, 1);
+            Assert.AreEqual(result.Errors.First().Code, "LIB002");
+        }
+
+        [TestMethod]
+        public async Task IsValidAsync_State_FileSystem_FilesDoNotExist()
+        {
+            var state = new Mocks.LibraryInstallationState
+            {
+                ProviderId = "filesystem",
+                LibraryId = "http://glyphlist.azurewebsites.net/img/images/Flag.png",
+                DestinationPath = "lib",
+                Files = new[] { "foo.png" }
+            };
+
+            ILibraryOperationResult result = await state.IsValidAsync(_dependencies);
+
+            Assert.IsTrue(result.Success);
```

Comment:
Could you add a comment to indicate why this is expected to succeed. 
Took me a while to recollect that FS provider supports renaming.